core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -1020,7 +1020,11 @@ object ZManaged {
               } yield b).ensuring((queue.shutdown *> ZIO.foreach_(fibers)(_.interrupt)).uninterruptible)
             }
           },
-          ZIO.flatten(finalizers.get)
+          for {
+            fs    <- finalizers.get
+            exits <- ZIO.foreach(fs)(_.run)
+            _     <- ZIO.done(Exit.collectAll(exits).getOrElse(Exit.unit))
```

Comment:
Let's use `Exit.collectAllPar` here (and in `reduceAllParN`) as that would better reflect the nature of errors that may have occured.

Reply:
All set.